library/alloc/src/vec/into_iter.rs

```diff
@@ -311,8 +314,8 @@ unsafe impl<#[may_dangle] T, A: Allocator> Drop for IntoIter<T, A> {
         impl<T, A: Allocator> Drop for DropGuard<'_, T, A> {
             fn drop(&mut self) {
                 unsafe {
-                    // `IntoIter::alloc` is not used anymore after this
-                    let alloc = ptr::read(&self.0.alloc);
+                    // `IntoIter::alloc` is not used anymore after this and will be dropped by RawVec
+                    let alloc = ManuallyDrop::into_inner(ptr::read(&self.0.alloc));
```

Comment:
Nit: `MD::take()` is a shorthand for `MD::into_inner(ptr::read())`, and one which conveys better the actual semantics involved (the `alloc` is conceptually "passed" from the `IntoIter` to the `DropGuard`):

```suggestion
                    let alloc = ManuallyDrop::take(&mut self.0.alloc);
```